tools/rostopic/src/rostopic/__init__.py

```diff
@@ -300,28 +300,37 @@ def msgevalgen(pattern):
 
     # use slice arguments if present
     array_index_or_slice_object = None
-    index = msg_attribute.find('[')
-    if index != -1:
-        if not msg_attribute.endswith(']'):
+    rec_msgeval = None
+    index_start = msg_attribute.find('[')
+    if index_start != -1:
+        index_end = msg_attribute.find(']')
+        if index_end == -1:
             sys.stderr.write("Topic name '%s' contains '[' but does not end with ']'\n" % msg_attribute)
             return None
-        index_string = msg_attribute[index + 1:-1]
+        index_string = msg_attribute[index_start + 1:index_end]
         try:
             array_index_or_slice_object = _get_array_index_or_slice_object(index_string)
         except AssertionError as e:
             sys.stderr.write("Topic name '%s' contains invalid slice argument '%s': %s\n" % (msg_attribute, index_string, str(e)))
             return None
-        msg_attribute = msg_attribute[:index]
+        msg_attribute_tail = msg_attribute[index_end+1:]
+        rec_msgeval = msgevalgen(msg_attribute_tail)
+        msg_attribute = msg_attribute[:index_start]
 
     def msgeval(msg):
         # I will probably replace this with some less beautiful but more efficient
         try:
             value = _get_nested_attribute(msg, msg_attribute)
         except AttributeError as e:
-            sys.stdout.write("no field named [%s]"%pattern+"\n")
+            if len(msg) > 1:
```

Comment:
Isn't this happening if you try to output e.g. `/tactile_mid_prox/sensors[1]/proximal`? Or if the message field is a string?